<launch>

  <arg name="publisher_uav_node_id" default="97"/>
  <arg name="subscriber_uav_node_id" default="98"/>
  <arg name="uav_can_interface" default="can0"/>

  <node name="ros_to_uavcan_bridge_node" pkg="sam_uavcan_bridge" type="sam_ros_to_uavcan_bridge" respawn="true" output="screen">
    <param name="uav_node_id" value="$(arg subscriber_uav_node_id)"/>
    <param name="uav_can_interface" value="$(arg uav_can_interface)"/>
    <remap from="~command" to="uavcan_command"/>
    <remap from="~array_command" to="uavcan_array_command"/>
    <remap from="~vbs_command" to="uavcan_vbs_command"/>
    <remap from="~tcg_command" to="uavcan_tcg_command"/>
    <remap from="~lcg_command" to="uavcan_lcg_command"/>
  </node>

  <node name="uavcan_to_ros_bridge_node" pkg="sam_uavcan_bridge" type="sam_uavcan_to_ros_bridge" respawn="false" output="screen">
    <param name="uav_node_id" value="$(arg publisher_uav_node_id)"/>
    <param name="uav_can_interface" value="$(arg uav_can_interface)"/>
    <remap from="~imu" to="uavcan_imu"/>
    <remap from="~pressure1" to="uavcan_pressure1"/>
    <remap from="~pressure2" to="uavcan_pressure2"/>
    <remap from="~leak" to="uavcan_leak"/>
    <remap from="~shallow_pressure" to="uavcan_shallow_pressure"/>
    <remap from="~tcg_feedback1" to="uavcan_tcg_feedback1"/>
    <remap from="~tcg_feedback2" to="uavcan_tcg_feedback2"/>
  </node>

</launch>
